haystack/pipelines/config.py

```diff
@@ -111,9 +127,70 @@ def build_component_dependency_graph(
     return graph
 
 
-def _validate_user_input(input: str):
-    if isinstance(input, str) and not VALID_CODE_GEN_INPUT_REGEX.match(input):
-        raise ValueError(f"'{input}' is not a valid config variable name. Use word characters only.")
+def validate_yaml(path: Path):
+    """
+    Validates the given YAML file using the autogenerated JSON schema.
+
+    :param pipeline_config: the configuration to validate
+    :return: None if validation is successful
+    :raise: `PipelineConfigError` in case of issues.
+    """
+    pipeline_config = read_pipeline_config_from_yaml(path)
+    validate_config(pipeline_config=pipeline_config)
+    logging.debug(f"'{path}' contains valid Haystack pipelines.")
+
+
+def validate_config(pipeline_config: Dict) -> None:
+    """
+    Validates the given configuration using the autogenerated JSON schema.
+
+    :param pipeline_config: the configuration to validate
+    :return: None if validation is successful
+    :raise: `PipelineConfigError` in case of issues.
+    """
+    validate_config_strings(pipeline_config)
+
+    with open(JSON_SCHEMAS_PATH / f"haystack-pipeline-unstable.schema.json", "r") as schema_file:
```

Comment:
For IDEs. They know where to find the schemas (thanks to SchemaStore) so when you write the version number, they'll pick the correct one from this folder. This enables versioned validation in the editors, which I think it's a great feature for stability.

Reply:
But you gave me a good idea: instead of packaging all the JSON schema with haystack, I'll package only unstable (which is what's used in the code anyway). The others can stay in the repo only, as they're needed only by SchemaStore. :+1: